src/state.rs

```diff
@@ -1,13 +1,21 @@
-use {super::*, num_traits::identities::Zero};
+use {
+  super::*,
+  num_traits::identities::Zero,
+  rand::{rngs::StdRng, SeedableRng},
+};
 
 pub(crate) struct State {
-  matrix: DMatrix<Vector3<u8>>,
+  pub(crate) matrix: DMatrix<Vector3<u8>>,
+  pub(crate) operation: Operation,
+  pub(crate) rng: StdRng,
 }
 
 impl State {
   pub(crate) fn new() -> Self {
     Self {
       matrix: DMatrix::zeros(0, 0),
+      operation: Operation::Invert,
+      rng: StdRng::seed_from_u64(0),
```

Comment:
Is there a reason we don't just call `rand::random::<u8>` in `Operation::apply`?

Reply:
Sorry, I should have explained ,it's a bit opaque. `rand::random()` uses the thread-local random number generator, which is seeded with external entropy on each run, making it produce a different sequence of values from run to run. I wanted to make the results of degenerate deterministic, so this adds a new rng to state. As long as we use this rng consistently, results will be consistent from run to run.

If we want to get random results, we can add a `seed` command that seeds `State::rng` with entropy.